r2/r2/lib/base.py

```diff
@@ -103,8 +112,8 @@ def __call__(self, environ, start_response):
         if cdn_ip:
             request.ip = cdn_ip
             request.via_cdn = True
-        elif g.trust_local_proxies and forwarded_for and is_local_address(remote_addr):
-            request.ip = forwarded_for.split(',')[-1]
+        elif g.trust_local_proxies and forwarded_for and is_trusted_proxy(remote_addr):
+            request.ip = forwarded_for.split(',')[0].strip()
```

Comment:
> extract correct IP when multiple IPs are present. The de facto spec[2] says that each proxy appends its peer's IP, which means the original requestor is the first IP, not the last.

The reason we pull the last IP is that it's the only one we can trust, which is the point of checking `is_trusted_proxy`. Anyone can put whatever they want in the XFF headers that travel over the internet.